src/traversing.js

```diff
@@ -168,10 +159,18 @@ jQuery.each({
 		return elem.contentDocument || jQuery.merge( [], elem.childNodes );
 	}
 }, function( name, fn ) {
+	var // Sorting (which implicitly de-dupes) is required for non-adjacent traversals
+		sort = name !== "prev" && name !== "next",
```

Comment:
Since `.children()` is getting a hit here already, it sounds like the UI performance issue is going to return regardless of what we do to `.next()` or `.prev()`. 

We are ignoring hints about node order here and I suspect we could do an insertion or merge sort with de-duping more efficiently than collecting the entire set of partially-ordered nodes and passing them on to `unique()`. But that would require more code, probably a lot more.